rmw_cyclonedds_cpp/src/rmw_node.cpp

```diff
@@ -3835,30 +3835,34 @@ static rmw_ret_t destroy_client(const rmw_node_t * node, rmw_client_t * client)
   RET_WRONG_IMPLID(node);
   RET_NULL(client);
   RET_WRONG_IMPLID(client);
-  auto info = static_cast<CddsClient *>(client->data);
   clean_waitset_caches();
 
-  {
-    // Update graph
-    auto common = &node->context->impl->common;
-    std::lock_guard<std::mutex> guard(common->node_update_mutex);
-    static_cast<void>(common->graph_cache.dissociate_writer(
-      info->client.pub->gid, common->gid,
-      node->name, node->namespace_));
-    rmw_dds_common::msg::ParticipantEntitiesInfo msg =
-      common->graph_cache.dissociate_reader(
-      info->client.sub->gid, common->gid, node->name,
-      node->namespace_);
-    if (RMW_RET_OK != rmw_publish(
-        common->pub,
-        static_cast<void *>(&msg),
-        nullptr))
+  if (client->data) {
```

Comment:
why this check?

Reply:
check the pointer before using it and also avoid double free for it